webcam-capture-drivers/driver-raspistill/src/main/java/com/github/sarxos/webcam/ds/raspistill/RaspistillDriver.java

```diff
@@ -0,0 +1,141 @@
+package com.github.sarxos.webcam.ds.raspistill;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Properties;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+import org.apache.commons.cli.PosixParser;
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.sarxos.webcam.WebcamDevice;
+import com.github.sarxos.webcam.WebcamDriver;
+
+/**
+ * 
+ * ClassName: RaspistillDriver <br/> 
+ * Function: driver of raspistill, launch raspistill process then communicate with it. this driver is special for raspberrypi <br/> 
+ * date: Jan 23, 2019 9:57:03 AM <br/> 
+ * 
+ * @author maoanapex88@163.com (alexmao86)
+ */
+public class RaspistillDriver implements WebcamDriver{
+	private final static Logger LOGGER=LoggerFactory.getLogger(RaspistillDriver.class);
+	private final Options options;
+	private Map<String, String> arguments=new LinkedHashMap<>();
+	/**
+	 * 
+	 * Creates a new instance of RaspistillDriver. 
+	 * step 1: load defaults.properties arguments
+	 * step 2: try to search current class loader's class path if raspistill.properties exits
+	 * step 3: search system properties
+	 */
+	public RaspistillDriver() {
+		options=OptionsBuilder.create();
+		
+		InputStream input=RaspistillDriver.class.getResourceAsStream("defaults.properties");
```

Comment:
I understand that loading settings from a file is useful but I don't know if it should be executed when the driver is initialized.
I know this is not a heavy processing but it is not a standard comparing to the other drivers.
But it's just my personal opinion.

Reply:
I will research this more, maybe one singleton instance is better